runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -524,9 +530,8 @@ public void sendResponseToServer(ChannelHandlerContext ctx, CorfuMsg inMsg, Corf
      */
     public void completeExceptionally(long requestID, Throwable cause) {
         CompletableFuture cf;
-        if ((cf = outstandingRequests.get(requestID)) != null) {
+        if ((cf = outstandingRequests.remove(requestID)) != null) {
```

Comment:
If cause can be null, then this changes the behavior by removing request first before completeExceptionally. if cause can not be null then add @Nonnull annotation.

Reply:
The caller of this is simply catching exception and passing it to this method, so cause is not null. Added @Nonnull annotation.

Reply:
`
if(outstandingRequests.contains(...)){
    ...
}
` ?

Reply:
@xnull remove is simpler, if you use contains, you need to call remove again.

Reply:
up to you but it looks awkward